scripts/core/ArticlesListV2.tsx

```diff
@@ -0,0 +1,164 @@
+
+    onItemClick(item: IArticle): void;
+    onItemDoubleClick?(item: IArticle): void;
+    multiSelect?: IMultiSelectNew;
+}
+
+export class ArticlesListV2 extends React.Component<IProps, IState> {
+    monitoringState: any;
+    lazyLoaderRef: LazyLoader<IArticle>;
+    removeContentUpdateListener: () => void;
+
+    constructor(props: any) {
+        super(props);
+
+        this.state = {
+            initialized: false,
+        };
+
+        this.monitoringState = ng.get('monitoringState');
+
+        this.loadMore = this.loadMore.bind(this);
+    }
+
+    loadMore(from: number, to: number) {
+        const {loadItems} = this.props;
+
+        return new Promise<Dictionary<string, IArticle>>((resolve) => {
+            loadItems(from, to).then((items) => {
+                const patch = items.reduce<{[key: string]: IArticle}>((acc, item, index) => {
+                    acc[from + index] = item;
+
+                    return acc;
+                }, {});
+
+                resolve(patch);
+            });
+        });
+    }
+
+    componentDidMount() {
+        this.monitoringState.init().then(() => {
+            this.setState({initialized: true});
+        });
+    }
+
+    componentWillUnmount() {
+        this.removeContentUpdateListener();
+    }
+
+    render() {
+        if (this.state.initialized !== true) {
+            return null;
+        }
+
+        const {itemCount, pageSize} = this.props;
+
+        return (
+            <LazyLoader
+                itemCount={itemCount}
+                loadMoreItems={this.loadMore}
+                pageSize={pageSize}
+                getId={(item) => item._id}
+                getItemsByIds={(ids) => {
+                    return Promise.all(
+                        ids.map((id) => dataApi.findOne<IArticle>('search', id)),
+                    );
+                }}
+                ref={(component) => {
+                    this.lazyLoaderRef = component;
+
+                    if (this.lazyLoaderRef != null && this.removeContentUpdateListener == null) {
+                        // wouldn't work in componentDidMount, because this.state.loading would be true
+                        // and LazyLoader wouldn't be mounted at that point yet.
+                        this.removeContentUpdateListener = addWebsocketEventListener(
+                            'resource:updated',
+                            ({extra}) => {
+                                if (
+                                    extra.resource === 'archive'
+                                    || extra.resource === 'archive_unspike'
+                                ) {
+                                    const reloadTheList = this.props?.shouldReloadTheList(
+                                        new Set(Array.from(Object.keys(extra.fields))),
+                                    ) ?? false;
+
+                                    if (reloadTheList) {
+                                        this.lazyLoaderRef.reset();
```

Comment:
would be good to do some throttling, there are some multi actions updating many items at the same time

Reply:
Good point. I've added it.